### PR TITLE
Breach in user profile jsp

### PR DESCRIPTION
**Pull request by sva605**
A couple of breaches related to userProfile have been fixed

### CODE REVIEW

File: web/src/main/java/ua/com/vertex/controllers/exceptionHandling/GlobalExceptionHandler.java [hunk 1]
```diff
@@ -50,6 +63,19 @@ public String handleCannotGetJdbcConnectionException(Exception e, Model model) {
         return ERROR;
     }
 
+    @ExceptionHandler(MultipartValidationException.class)
+    public String handleMultipartValidationException(MultipartValidationException e, Model model) {
+        if (FILE_SIZE_EXCEEDED.equals(e.getMessage())) {
```
Comment (backumich) [inline web/src/main/java/ua/com/vertex/controllers/exceptionHandling/GlobalExceptionHandler.java]: can you please add readable message to exception and after that just add e.getMessage() as error message to model? Without if-else blocks

File: web/src/main/java/ua/com/vertex/controllers/exceptionHandling/exceptions/MultipartValidationException.java [hunk 2]
```diff
@@ -0,0 +1,7 @@
+package ua.com.vertex.controllers.exceptionHandling.exceptions;
+
+public class MultipartValidationException extends RuntimeException {
+    public MultipartValidationException(String message) {
```
Comment (backumich) [inline web/src/main/java/ua/com/vertex/controllers/exceptionHandling/exceptions/MultipartValidationException.java:4]: please add 'Throwable cause' parameter

File: web/src/main/java/ua/com/vertex/controllers/ImageController.java [hunk 3]
```diff
@@ -7,57 +7,74 @@
...
+        encode(model, user.getEmail(), imageType);
+        model.addAttribute(USER, user);
+        LOGGER.debug("Passing imagePassport to JSP");
+
+        return IMAGE_PASSPORT;
+    }
+
+    private void encode(Model model, String email, String imageType) {
+        String encoded = Base64.encodeBase64String(userLogic.getImage(email, imageType).orElse(new byte[]{}));
         model.addAttribute(imageType, encoded);
```
Comment (backumich) [inline web/src/main/java/ua/com/vertex/controllers/ImageController.java]: model is indirectly changed. Can you please get encoded image from the method and set it to the model inside interface method.

**Review by backumich — CHANGES_REQUESTED** [review]

File: web/src/main/java/ua/com/vertex/controllers/ImageController.java (hunk 3 above)
Comment (sva605) [reply web/src/main/java/ua/com/vertex/controllers/ImageController.java]: Done

File: web/src/main/java/ua/com/vertex/controllers/exceptionHandling/GlobalExceptionHandler.java (hunk 1 above)
Comment (sva605) [reply web/src/main/java/ua/com/vertex/controllers/exceptionHandling/GlobalExceptionHandler.java]: Done

File: web/src/main/java/ua/com/vertex/controllers/exceptionHandling/exceptions/MultipartValidationException.java (hunk 2 above)
Comment (sva605) [reply web/src/main/java/ua/com/vertex/controllers/exceptionHandling/exceptions/MultipartValidationException.java:4]: I don't exactly understand what you want here

Comment (backumich) [reply web/src/main/java/ua/com/vertex/controllers/exceptionHandling/exceptions/MultipartValidationException.java:4]: I mean adding one more method parameter with type 'Throwable' and name 'cause' - exception that would be mentioned as 'caused by' in stacktrace.

File: web/src/main/java/ua/com/vertex/logic/UserLogicImpl.java [hunk 4]
```diff
@@ -68,9 +67,10 @@ public void saveImage(String email, MultipartFile file, String imageType) {
 
     private void validateMultipartFile(MultipartFile file) {
         if (file.getSize() > fileSizeInBytes) {
-            throw new MultipartValidationException(FILE_SIZE_EXCEEDED);
+            throw new MultipartValidationException("File size exceeded, max allowed is " +
+                    UtilFunctions.humanReadableByteCount(fileSizeInBytes));
         } else if (!file.getContentType().split("/")[0].equals("image")) {
-            throw new MultipartValidationException(FILE_TYPE_INVALID);
+            throw new MultipartValidationException("You have chosen a file of invalid type!");
```
Comment (backumich) [inline web/src/main/java/ua/com/vertex/logic/UserLogicImpl.java]: can you please be more polite with customer? Without '!' !!!!!!!!!!!!!!!!!!!!!!! =)

**Review by backumich — CHANGES_REQUESTED** [review]

File: web/src/main/java/ua/com/vertex/controllers/exceptionHandling/exceptions/MultipartValidationException.java (hunk 2 above)
Comment (backumich) [reply web/src/main/java/ua/com/vertex/controllers/exceptionHandling/exceptions/MultipartValidationException.java:4]: Actually, I've took a look on it and it's fine for now. No additional parameters are needed

**Review by backumich — APPROVED** [review]

**Pull request merged**